### PR TITLE
Add catalog CRD key functions

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/roadmap/issues/219

### CODE REVIEW

File: pkg/key/catalog.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package key
+
+import (
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/microerror"
+)
+
+func CatalogTitle(customResource v1alpha1.Catalog) string {
```
Comment (tomahawk28) [inline pkg/key/catalog.go:9]: I have changed all `AppCatalog*` to `Catalog` in this file.

File: pkg/key/catalog.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	return customResource.Spec.Title
+}
+
+func CatalogStorageURL(customResource v1alpha1.Catalog) string {
+	return customResource.Spec.Storage.URL
+}
+
+func CatalogConfigMapName(customResource v1alpha1.Catalog) string {
+	return customResource.Spec.Config.ConfigMap.Name
```
Comment (rossf7) [inline pkg/key/catalog.go:18]: The config struct and fields below are optional. So should we add nil checks?

Components using the key funcs then don't need to worry about this. WDYT?

Comment (tomahawk28) [reply pkg/key/catalog.go:18]: That's nice, I think we can return an empty string in that case, right? 
I will do that :) Thanks for your suggestion!

Comment (rossf7) [reply pkg/key/catalog.go:18]: Yes, we can return empty strings. Thx!

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**